demo/src/example/example-page.js

```diff
@@ -8,67 +8,215 @@ import router from '../core/router';
+    const vendor = this.#drmVendorEl.value;
+
+    if (!vendor) {
+      return undefined;
+    }
+
+    const keySystems = {};
+    const certificateUrl = this.#certificateUrlEl.value;
+    const licenseUrl = this.#licenseUrlEl.value;
+
+    if ('com.apple.fps.1_0' === vendor) {
+      keySystems[vendor] = { certificateUrl, licenseUrl };
+    } else {
+      keySystems[vendor] = licenseUrl;
+    }
+
+    return keySystems;
   }
 
   /**
-   * Handles the submission of a source (URL or URN).
+   * Get the source configured in the load bar.
    *
-   * @param {string} src - The submitted source (URL or URN).
+   * @returns {string} The source configured by the user.
    */
-  onSourceSubmitted(src) {
+  get customSrc() {
+    return this.#loadBarEl.value;
+  }
+
+  /**
+   * Play the currently configured source and DRM settings in the player modal.
+   */
+  playCustomUrl() {
+    const src = this.customSrc;
     const type = src.startsWith('urn:') ? 'srgssr/urn' : undefined;
+    const keySystems = this.keySystems;
+
+    openPlayerModal({ src, type, keySystems });
+  }
+
+  /**
+   * Initializes the DRM settings element.
+   */
+  initDrmSettings() {
+    const settingsEl = document.querySelector('#load-bar-settings');
+    const settingsBtnEl = document.querySelector('#open-settings-btn');
+    const settingsIconEl = settingsBtnEl.querySelector('span');
+
+    settingsEl.addEventListener('animationend', () => {
+      settingsEl.classList.toggle('fade-in', false);
+    });
+
+    settingsIconEl.addEventListener('animationend', () => {
+      settingsIconEl.classList.toggle('spin', false);
+      settingsIconEl.classList.toggle('spin-back', false);
+    });
+
+    // Add event listener for the 'click' event on the '#open-settings-btn' element.
+    settingsBtnEl.addEventListener('click', () => {
+      settingsEl.classList.toggle('hidden');
+      const isHidden = settingsEl.classList.contains('hidden');
 
-    openPlayerModal({ src, type });
+      settingsEl.classList.toggle('fade-in', !isHidden);
+      settingsIconEl.classList.toggle('spin', !isHidden);
+      settingsIconEl.classList.toggle('spin-back', isHidden);
+      settingsEl.ariaHidden = isHidden.toString();
+    });
+  }
+
+  /**
+   * The HTML template of the load bar.
+   */
+  loadBarTemplate() {
+    return `
+      <div class="fade-in">
+        <div class="load-bar-container">
+          <span class="material-icons-outlined">insert_link</span>
+          <input type="text" id="load-bar" placeholder="Enter a URL or URN to play its content...">
+          <button id="open-settings-btn" title="Open DRM Settings"><span class="material-icons-outlined">settings</span></button>
+        </div>
+        
+        <form id="load-bar-settings" class="drm-settings-container hidden" aria-hidden="true">
+          <h3>DRM Settings</h3>
+          <select id="drm-vendor" aria-label="Select a DRM vendor" required>
+            <option value="" disabled selected>Select a DRM vendor</option>
```

Comment:
pls see my comment https://github.com/SRGSSR/pillarbox-web/pull/96/files#r1402162348

Reply:
This is my answer : https://github.com/SRGSSR/pillarbox-web/pull/96#discussion_r1402177729 !